app/controllers/tracks/concepts_controller.rb

```diff
@@ -0,0 +1,47 @@
+class Tracks::ConceptsController < ApplicationController
+  before_action :use_track
+
+  allow_unauthenticated! :index, :show
+
+  def authenticated_index
+    use_concepts
+
+    if current_user.joined_track?(@track)
```

Comment:
Okay with this for now. But I'm thinking to redirect to another action when it's unjoined? Maybe something like `tracks/concepts/preview`. Just feel like it's more concrete that way.

Reply:
I don't want the URLs to be different though, which is why this whole (slightly painful) setup is in place, otherwise I'd just use normal methods.

I want people to have consistent URLs whether they're Google, joined, not joined, etc.